src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Component.targets

```diff
@@ -85,7 +86,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     Name="RazorGenerateComponentDeclaration"
     DependsOnTargets="$(RazorGenerateComponentDeclarationDependsOn)"
     Inputs="$(MSBuildAllProjects);@(RazorComponentWithTargetPath);$(_RazorComponentInputCacheFile)"
-    Outputs="@(_RazorComponentDeclaration)"
+    Outputs="$(_RazorComponentDeclarationOutputCacheFile)"
```

Comment:
Prior to this, we relied on the declaration files being older than the inputs to determine if the targets need to be executed. This now uses an explicit marker file. This allows declaration files to not be updated, but for us to track if we at least tried regenerating the declaration file.

Reply:
What do the contents of this declaration file look like? Does the `RazorGenerateComponentDeclaration` task need to be changed?

Reply:
The declaration files are the outline of the components with out any body (they kinda look like ref assemblies). It lets us know what components exist and what parameters are on them. Prior to this change, if you were editing `Index.razor` to change some markup, here's what the sequence of events would look like:

```
RazorGenerateComponentDeclaration
    Inputs: Index.razor
    Output: obj\Debug\net5.0\RazorDeclaration\Index.razor.g.cs

MSBuild would evaluate this target because `Index.razor` is newer than `Index.razor.g.cs` (since it was just edited). At the end of this target, `Index.razor.g.cs` would have a new timestamp but the contents would be unchanged.

Next up is the compile and tag helper discovery targets that we use to find out what changed in the compnents:

RazorCompileComponentDeclaration
    Inputs: obj\Debug\net5.0\RazorDeclaration\Index.razor.g.cs
    Output: obj\Debug\net5.0\RazorDeclaration\Assembly.declaration.dll

Since `Index.razor.g.cs` is newer, MSBuild will evaluate this target and perform CSC. Next up, since the declaration assembly is newer, we'll perform a tag helper discovery on it.

```


```
RazorGenerateComponentDeclaration
    Inputs: Index.razor
    Output: SomeMarkerFile

MSBuild would evaluate this target because `Index.razor` is newer than `SomeMarkerFile`. At the end of this target, `Index.razor.g.cs` would have not a have newer timestamp. This allows the next two targets to be skipped. SomeMarkerFile helps us track if the target was evaluated since the inputs were last changed.
```